nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/MoveHDFS.java

```diff
@@ -228,14 +232,15 @@ public void onScheduled(ProcessContext context) throws Exception {
 
     @Override
     public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
-        // MoveHDFS
-        FlowFile parentFlowFile = session.get();
-        if (parentFlowFile == null) {
-            return;
+        FlowFile flowFile = null;
+
+        if (context.hasIncomingConnection()) {
+            flowFile = session.get();
```

Comment:
If we have an incoming relationship and we call session.get() and get back a null flow file, then I think we want to return. We only want to create a new flow file if there is no incoming relationship.

Reply:
Done